x-pack/plugins/security_solution/scripts/endpoint/agent_downloader_cli/agent_downloader.ts

```diff
@@ -11,21 +11,63 @@ import type { ToolingLog } from '@kbn/tooling-log';
 import { getAgentDownloadUrl, getAgentFileName } from '../common/fleet_services';
 import { downloadAndStoreAgent } from '../common/agent_downloads_service';
 
+// Decrement the patch version by 1
+const decrementPatchVersion = (version: string): string => {
+  const [major, minor, patch] = version.split('.').map(Number);
+  return `${major}.${minor}.${patch - 1}`;
+};
+
+// Generate a list of versions to attempt downloading, including a fallback to the previous patch (GA)
+const getVersionsToDownload = (version: string): string[] => {
+  const patch = parseInt(version.split('.')[2], 10);
+
+  // If patch version is 0, return only the current version.
+  if (patch === 0) {
+    return [version];
+  }
+
+  return [version, decrementPatchVersion(version)];
+};
+// Download and store the Elastic Agent for the specified version(s)
 const downloadAndStoreElasticAgent = async (
   version: string,
   closestMatch: boolean,
   log: ToolingLog
-) => {
-  const downloadUrlResponse = await getAgentDownloadUrl(version, closestMatch, log);
-  const fileNameNoExtension = getAgentFileName(version);
-  const agentFile = `${fileNameNoExtension}.tar.gz`;
-  await downloadAndStoreAgent(downloadUrlResponse.url, agentFile);
+): Promise<void> => {
+  const versionsToDownload = getVersionsToDownload(version);
+
+  for (const v of versionsToDownload) {
+    try {
+      const { url } = await getAgentDownloadUrl(v, closestMatch, log);
+      const fileName = `${getAgentFileName(v)}.tar.gz`;
+
+      await downloadAndStoreAgent(url, fileName);
+      log.info(`Successfully downloaded and stored version ${v}`);
+      return; // Exit once successful
+    } catch (error) {
+      log.error(`Failed to download or store version ${v}: ${error.message}`);
```

Comment:
Just want to ensure you mean to ONLY log an error here and then keep going in downloading other versions. is that right?

Reply:
Yes, that's correct. If the first attempt fails, we move on to try downloading the fallback version.